share/shwap/pb/shwap.proto

```diff
@@ -26,11 +26,15 @@ message RowNamespaceData {
     proof.pb.Proof proof = 2;
 }
 
+message NamespaceData {
+    repeated RowNamespaceData namespaceData = 1;
+}
+
```

Comment:
What about also adding `RangeNamespaceData` in proto? It would embed `NamespaceData`. The main motivation would be keeping all the containers part of proto "spec" for consistency and clarity.